### PR TITLE
Fixed filtering logic for install workflow

### PR DESCRIPTION
**Pull request by cloudbehl**

### CODE REVIEW

**Comment by afreen23** [comment]
Its still buggy:
1.The `selection node count` is not changing
2. The selection of ticks is not happening, one tick gets removed on selection of other

**Comment by cloudbehl** [comment]
@afreen23 can you do a verification once again. There is still a minor glitch with selecting the filter dropdown. But the values are persistent.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx [hunk 1]
```diff
@@ -136,39 +136,42 @@ const stateToProps = (obj, { data = [], filters = {}, staticFilters = [{}] }) =>
...
   const [selectedNodesCnt, setSelectedNodesCnt] = React.useState(0);
 
   let storageClass = '';
 
   React.useEffect(() => {
-    const selectedNode = _.filter(nodes, 'selected').length;
+    const selectedNode = _.filter(unfilteredNodes, 'selected').length;
     setSelectedNodesCnt(selectedNode);
-  }, [nodes]);
+  }, [unfilteredNodes]);
```
[the remark this answers is not in the document]
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: @gnehapk ^^

**Comment by afreen23** [comment]
> @afreen23 can you do a verification once again. There is still a minor glitch with selecting the filter dropdown. But the values are persistent.

The selection thing needs to be fixed though !
Also, please update the description with the bug that this patch fixes.

**Comment by gnehapk** [comment]
/approve
@afreen23 Can you also test it one?

**Comment by gnehapk** [comment]
@cloudbehl Please rebase your PR.

File: frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx [hunk 2]
```diff
@@ -190,6 +195,13 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({ data, loaded, ocsProp
       formattedNodes[index].selected = isSelected;
     }
     setNodes(formattedNodes);
+    const nodesByID = _.keyBy(nodes, 'id');
+    const setSelectedUnfilteredNodes = _.each(unfilteredNodes, (n) => {
+      if (_.get(nodesByID, [n.id, 'id']) === n.metadata.name) {
+        n.selected = _.get(nodesByID, [getName(n), 'selected'], false);
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx]: you are not importing getName , its causing test to fail.

**Pull request merged**